gen/main.go

```diff
@@ -169,10 +169,10 @@ func writeGoTemplate(tmpls *template.Template, tmpl, outputPath string, tmplArg
 	} else {
 		data = formatted
 	}
-	if err := os.MkdirAll(filepath.Dir(outputPath), 0755); err != nil {
+	if err := os.MkdirAll(filepath.Dir(outputPath), 0700); err != nil {
 		return errs.Wrap(err)
 	}
-	if err := ioutil.WriteFile(outputPath, data, 0644); err != nil { //nolint:gosec // these are the permissions I want
+	if err := ioutil.WriteFile(outputPath, data, 0600); err != nil { //nolint:gosec // these are the permissions I want
```

Comment:
This (and the permissions above for the mkdir as well) is creating Go source files, potentially for inclusion in source control. I don't want these to be super-restrictive by default and would prefer them to retain their old values.